BookA-Pset_GeoObsMappedUnit

larswik opened the pull request:
# Pset: Pset_GeoObsMappedUnit

## Applicability: IfcGeoScienceObservation/MAPPEDFEATURE

## Description: 

Properties for geoscience observations at mapped units

| Property | Datatype | Description |
|----------|----------|-------------|
|AreaID|IfcLabel|Indication of the single, specific MappedUnit in the same Mapped Zone|
|CoveragePercentage|IfcPositiveRatioMeasure|Related to the MappedZone, e.g.% of the tunnel face where a certian ground types was documented|
|Weathering|IfcLabel|Description of the physical and chemical changes produced by atmospheric agents in rocks or other deposits at or near the earth's surface. These changes result in disintegration and de composition of the material.|
|DiscontinuitySpacing|IfcLabel|Discontinuity Spacing: The term "spacing" refers to the mean or modal spacing of a set of discontinuities and is the perpendicular distance between adjacent discontinuities. The discontinuity spacing should be measured in millimetres and can be classified using the terms in Table 8|
|GSI|IfcInteger|The Geological Strength Index (GSI): A characterization tool for assessing engineering properties for rock masses. >NOTE See also https://static.rocscience.cloud/assets/resources/learning/hoek/2000-GSI-A-Geologically-Friendly-Tool-for-Rock-Mass-Strength-Estimation.pdf|
|RMR|IfcInteger|The rock mass rating (RMR) is a geomechanical classification system for rocks, developed by Z. T. Bieniawski. >NOTE See also ASTM (1988). "Standard Guide for using the Rock Mass Rating (RMR) System (Geomechanics Classification) in Engineering Practices." American Society for Testing and Materials, Book of Standards D5878-08, v.04.09, Philadelphia, PA.|
|QValue|IfcReal|A description of the rock mass stability of an underground opening in jointed rock masses based on Q-system which was originaly developed at NGI. 2015 - Handbook on the Q-System - NGI. >NOTE See also Barton, N.R.; Lien, R.; Lunde, J. (1974). "Engineering classification of rock masses for the design of tunnel support". Rock Mechanics and Rock Engineering.|

larswik committed:
BookA-Pset_GeoObsMappedUnit

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

larswik committed:
Update Documentation.md

SergejMuhic commented:
Descriptions of the last three properties require references?

j-weil commented:
GSI: https://static.rocscience.cloud/assets/resources/learning/hoek/2000-GSI-A-Geologically-Friendly-Tool-for-Rock-Mass-Strength-Estimation.pdf
RMR: ASTM (1988). "Standard Guide for using the Rock Mass Rating (RMR) System (Geomechanics Classification) in Engineering Practices." American Society for Testing and Materials, Book of Standards D5878-08, v.04.09, Philadelphia, PA.
Or https://en.wikipedia.org/wiki/Rock_mass_rating (i Wikipedia is a vaild reference? It gives the best free online overview
Qvalue: Barton, N.R.; Lien, R.; Lunde, J. (1974). "Engineering classification of rock masses for the design of tunnel support". Rock Mechanics and Rock Engineering. [Springer](https://en.wikipedia.org/wiki/Springer_Science%2BBusiness_Media).
or https://en.wikipedia.org/wiki/Q-system_(geotechnical_engineering)

larswik commented:
I saw another thing now: What does "Table 8" refer to in the definition of DiscontinuitySpacing? Ping @j-weil

larswik commented on IFC4x3/Properties/g/GSI_0jXoIJ8Fz6WuI01oi0uMuh/Documentation.md:
```diff
@@ -0,0 +1 @@
+The Geological Strength Index (GSI): A characterization tool for assessing engineering properties for rock masses.
```
```suggestion
The Geological Strength Index (GSI): A characterization tool for assessing engineering properties for rock masses.
  >NOTE Definition from Marinos & Hoek (2000) GSI: A Geologically Friendly Tool For Rock Mass Strength Estimation. [Link](https://static.rocscience.cloud/assets/resources/learning/hoek/2000-GSI-A-Geologically-Friendly-Tool-for-Rock-Mass-Strength-Estimation.pdf)
```

larswik commented on IFC4x3/Properties/r/RMR_0P4g7RHODAnPtwhbwtKXgn/Documentation.md:
```diff
@@ -0,0 +1 @@
+The rock mass rating (RMR) is a geomechanical classification system for rocks, developed by Z. T. Bieniawski.
```
```suggestion
The rock mass rating (RMR) is a geomechanical classification system for rocks.
>NOTE Definition from ASTM (1988). Standard Guide for using the Rock Mass Rating (RMR) System (Geomechanics Classification) in Engineering Practices. American Society for Testing and Materials, Book of Standards D5878-08, v.04.09, Philadelphia, PA.
```

larswik commented on IFC4x3/Properties/q/QValue_2AP24nbKP3hfI8vdx$uG8T/Documentation.md:
```diff
@@ -0,0 +1 @@
+"A description of the rock mass stability of an underground opening in jointed rock masses based on Q-system which was originaly developed at NGI. 2015 - Handbook on the Q-System - NGI".
```
```suggestion
A description of the rock mass stability of an underground opening in jointed rock masses.
>NOTE Definition from Barton, N.R.; Lien, R.; Lunde, J. (1974). Engineering classification of rock masses for the design of tunnel support. Rock Mechanics and Rock Engineering.
```

j-weil commented:
"Table 8" in discontinuity spacing refers to ISO 14689, the ref for description of discontinuities in rock mass
![image](https://github.com/bSI-InfraRoom/IFC-Specification/assets/49076658/13045aaf-309b-4b1b-b81e-d19fb9a63028)
Please add ""Table 8 in ISO 14689"

larswik committed:
Update IFC4x3/Properties/g/GSI_0jXoIJ8Fz6WuI01oi0uMuh/Documentation.md

larswik committed:
Update IFC4x3/Properties/q/QValue_2AP24nbKP3hfI8vdx$uG8T/Documentation.md

larswik committed:
Update IFC4x3/Properties/r/RMR_0P4g7RHODAnPtwhbwtKXgn/Documentation.md

larswik commented on IFC4x3/Properties/d/DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e/Documentation.md:
```diff
@@ -0,0 +1 @@
+Discontinuity Spacing: The term "spacing" refers to the mean or modal spacing of a set of discontinuities and is the perpendicular distance between adjacent discontinuities. The discontinuity spacing should be measured in millimetres and can be classified using the terms in Table 8.
```
```suggestion
Discontinuity Spacing: The term "spacing" refers to the mean or modal spacing of a set of discontinuities and is the perpendicular distance between adjacent discontinuities. The discontinuity spacing should be measured in millimetres..
>NOTE Definition from ISO 14689:2017, Table 8.
```

larswik committed:
Update IFC4x3/Properties/d/DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e/Documentation.md

SergejMuhic commented on IFC4x3/Properties/a/AreaID_1LX3rTqy19RRHx3VNZyWTI/Documentation.md:
```diff
@@ -0,0 +1 @@
+Indication of the single, specific MappedUnit in the same Mapped Zone.
```
What is `MappedUnit`? What is `Mapped Zone`? Why are the formatted differently? It seems to me that they are not IFC based identifiers. Could we get a refence?

larswik replied:
I would write "Identification of the single, specific mapped feature in the same mapped zone."
And also add a note: ">NOTE Mapped zone is represented in IFC using _IfcSpatialZone_ using PredefinedType _MAPPEDZONE_."
I would also change datatype to IfcIdentifier as @SergejMuhic suggests below.
Ping @j-weil

j-weil replied:
IfcIdentifier sounds reasonable to me

SergejMuhic commented on IFC4x3/Properties/a/AreaID_1LX3rTqy19RRHx3VNZyWTI/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AreaID_1LX3rTqy19RRHx3VNZyWTI" Name="AreaID" UniqueId="55843d5d-d3c0-496d-b47b-0df5e3f20752" PrimaryDataType="IfcLabel" />
```
How about `IfcIdentifier`? Does it make sense or is it purely human readable?

larswik replied:
See previous comment.

SergejMuhic commented on IFC4x3/Properties/c/CoveragePercentage_0zErkL8B54XBTK$nUEKqm5/Documentation.md:
```diff
@@ -0,0 +1 @@
+Related to the MappedZone, e.g.% of the tunnel face where a certian ground types was documented.
```
`MappedZone` is formatted differently than in the `AreaID` description.

larswik replied:
Suggestion: Rewrite to "Coverage in percentage of the mapped feature in relation to the mapped zone, e.g. tunnel face, to which it belongs."
Ping @SergejMuhic and @j-weil.

SergejMuhic commented on IFC4x3/Properties/c/CoveragePercentage_0zErkL8B54XBTK$nUEKqm5/Documentation.md:
```diff
@@ -0,0 +1 @@
+Related to the MappedZone, e.g.% of the tunnel face where a certian ground types was documented.
```
```suggestion
Related to the MappedZone, e. g. percent of the tunnel face where a certain ground type was documented.
```

larswik replied:
See suggestion in previous comment.

SergejMuhic commented on IFC4x3/Properties/d/DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e" Name="DiscontinuitySpacing" UniqueId="1f6a32d2-7e8c-4abe-8aa1-3c1def44c268" PrimaryDataType="IfcLabel" />
```
Datatype does not seem to be correct. Referring to the description "perpendicular distance between adjacent discontinuities".

larswik replied:
Agreed! Shouldn't it be IfcPositiveLengthMeasure?
Ping @j-weil

j-weil replied:
Generally correct, but the Disc Spacing according to ISO 14689 is given in fixed ranges, according to the table above, i.e. predefined classes. We can either keep Ifclabel or define enumerations according to the second row of the table (Less han 20mm, 20 to 60mm, 60 to 200mm,...)

larswik replied:
Can't we make an enum out of the terms (first column) in Table 8? Ping @j-weil

j-weil replied:
I thought about it and think the second is better, becouse then we do not need a "legend" to check what was "very close" etc. In my opinion the second is the better and more common one that needs to be exchanged

j-weil replied:
answereing here on your email, Lars:
Can you check the last comment regarding Pset_GeoObsMappedUnit.DiscontinuitySpacing?
Alternatives:
-	Keep IfcLabel and refer to Table 8 first column
-	Keep IfcLabel and refer to Table 8 second column
-	Use the terms (first column) from Table 8 and create an enum?
**-	Use the Spacing mm (second column) from Table 8 and create an enum?** Jonas: This is the best option.

larswik replied:
After discussion with @SergejMuhic I changed so we use Table 8 column 1 since the enum typically should be more descriptive in its nature. Column 2 is used as descriptions for the constants.

SergejMuhic commented on IFC4x3/Properties/d/DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e/Documentation.md:
```diff
@@ -0,0 +1,2 @@
+Discontinuity Spacing: The term "spacing" refers to the mean or modal spacing of a set of discontinuities and is the perpendicular distance between adjacent discontinuities. The discontinuity spacing should be measured in millimetres..
```
```suggestion
Discontinuity Spacing: The term "spacing" refers to the mean or modal spacing of a set of discontinuities and is the perpendicular distance between adjacent discontinuities. The discontinuity spacing should be measured in millimetres.
```

SergejMuhic commented on IFC4x3/Properties/g/GSI_0jXoIJ8Fz6WuI01oi0uMuh/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="GSI_0jXoIJ8Fz6WuI01oi0uMuh" Name="GSI" UniqueId="2d872493-20ff-4683-8480-072b00e16e2b" PrimaryDataType="IfcInteger" />
```
Could we double check whether it is an integer?

larswik replied:
Seems plausible with IfcInteger when looking here:
![image](https://github.com/bSI-InfraRoom/IFC-Specification/assets/25506404/87068b1e-2608-4d35-88aa-1bf620cfb7fd)
Ping @j-weil

j-weil replied:
Generally correct, GSI value is always an integer. However, again a tricky one: The common practice is to provide the GSI as a range like e.g. 50-60 or 55-60 and not as an exact number. So IfcPropertyBoundedValue might be the best option.

larswik replied:
IfcpropertyBoundedValue it is then!

larswik replied:
Done!

SergejMuhic commented on IFC4x3/Properties/r/RMR_0P4g7RHODAnPtwhbwtKXgn/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="RMR_0P4g7RHODAnPtwhbwtKXgn" Name="RMR" UniqueId="1912a1db-4583-4ac5-9dfa-ae5eb7521ab1" PrimaryDataType="IfcInteger" />
```
Also double check whether integer.

j-weil replied:
Integer is OK for RMR

SergejMuhic commented on IFC4x3/Properties/w/Weathering_3VudkS3GP7whCc9fnu5Tf9/Documentation.md:
```diff
@@ -0,0 +1 @@
+Description of the physical and chemical changes produced by atmospheric agents in rocks or other deposits at or near the earth's surface. These changes result in disintegration and de composition of the material.
```
```suggestion
Description of the physical and chemical changes produced by atmospheric agents in rocks or other deposits at or near the earth's surface. These changes result in disintegration and decomposition of the material.
```

SergejMuhic commented on IFC4x3/Properties/w/Weathering_3VudkS3GP7whCc9fnu5Tf9/DocProperty.xml:
```diff
@@ -0,0 +1,2 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Weathering_3VudkS3GP7whCc9fnu5Tf9" Name="Weathering" UniqueId="dfe27b9c-0d06-47ea-b326-269c7815da49" PrimaryDataType="IfcLabel" />
```
If it is supposed to be a description, maybe consider `IfcText`.

larswik replied:
Ping @j-weil

j-weil replied:
Weathering should reference ISO 14689:2017(E) and be provided according to the table below:
![image](https://github.com/bSI-InfraRoom/IFC-Specification/assets/49076658/56564911-d27d-4860-b876-43fa4c51558d)
Column "Term" as predefined enumerations and adding the ref to ISO standard would be perfect

larswik replied:
Done!

larswik committed:
Update IFC4x3/Properties/d/DiscontinuitySpacing_0VQZBIVenAlegXF1tlHC9e/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Update IFC4x3/Properties/w/Weathering_3VudkS3GP7whCc9fnu5Tf9/Documentation.md

Co-authored-by: SergejMuhic <sergejs1@gmail.com>

larswik committed:
Merge remote-tracking branch 'upstream/tunnel' into feature/BookA-Pset_GeoObsMappedunit

larswik committed:
Merge branch 'feature/BookA-Pset_GeoObsMappedunit' of https://github.com/larswik/IFC-Specification into feature/BookA-Pset_GeoObsMappedunit

larswik committed:
According to comments from @SergejMuhic and @j-weil.

- Updated descriptions
- AreaID as IfcIdentifier
- DiscontinuitySpacing as PEnum
- Weathering as PEnum
- GSI as bounded value (IfcInteger)

larswik committed:
Changed enum constants for PEnum_DiscontinuitySpacing

Use Table 8, column one as constants and adding column 2 as descriptions for the constants.

larswik commented:
> "Table 8" in discontinuity spacing refers to ISO 14689, the ref for description of discontinuities in rock mass ![image](https://user-images.githubusercontent.com/49076658/261962846-13045aaf-309b-4b1b-b81e-d19fb9a63028.png) Please add ""Table 8 in ISO 14689"

This one is done.
